website/docs/sql-reference/window-functions/sql-row-number.md

```diff
@@ -0,0 +1,74 @@
+---
+id: row-number
+title: SQL ROW_NUMBER
+description: The ROW_NUMBER function returns the unique row number of a row in an ordered group or dataset.
+slug: /sql-reference/row-number
+---
+
+<head>
+    <title>Working with the SQL ROW_NUMBER</title>
+</head>
+
+In this page, let’s go deep into the ROW_NUMBER function and talk about what it is, how to use it, and why it’s important in analytics engineering work.
+
+The ROW_NUMBER window function is an effective way to create a ranked column or filter a query based on rankings. More specifically, the ROW_NUMBER function returns the *unique* row number of a row in an ordered group or dataset. 
+
+Unlike the [RANK](/sql-reference/rank) and DENSE_RANK functions, ROW_NUMBER is non-deterministic, meaning that a *unique* number is assigned arbitrarily for rows with duplicate values.
+
+## How to use the ROW_NUMBER function
+
+The ROW_NUMBER function has a pretty simple syntax, with an optional partition field and support for ordering customization:
+
+`row_number() over ([partition by <field(s)>] order by field(s) [asc | desc])`
+
+Some notes on this function’s syntax:
+
+- The `partition by` field is optional; if you want to get the row numbers of your entire dataset by certain fields (compared to partitioning *and grabbing row number* within a dataset), you would simply omit the `partition by` from the function call (see the example below for this).
+- By default, the ordering of a ROW_NUMBER function is set to ascending. To explicitly make the resulting order descending, you’ll need to pass in `desc` to the `order by` part of the function.
+
+Let’s take a look at a practical example using the ROW_NUMBER function below.
+
+### ROW_NUMBER function example
+
+```sql
+select
+    customer_id,
+    order_id,
+    order_date,
+    row_number() over (partition by customer_id order by order_date) as rnk
+from {{ ref('orders') }}
+order by 1
+```
+
+This simple query using the [Jaffle Shop’s](https://github.com/dbt-labs/jaffle_shop) `orders` table will return the unique row number per customer by their `order_date`:
+
+| customer_id | order_id | order_date | rnk |
+|:---:|:---:|:---:|:---:|
+| 1 | 1 | 2018-01-01 | 1 |
+| 1 | 37 | 2018-02-10 | 2 |
+| 2 | 8 | 2018-01-11 | 1 |
+| 3 | 2 | 2018-01-02 | 1 |
+| 3 | 24 | 2018-01-27 | 2 |
+| 3 | 69 | 2018-03-11 | 3 |
+
+Because ROW_NUMBER is non-deterministic, orders per customer that have the same `order_date` would have unique `rnk` values (unlike if you used the RANK or DENSE_RANK functions).
```

Comment:
```suggestion
Because ROW_NUMBER is non-deterministic, orders per customer that have the same `order_date` would have unique `row_n` values (unlike if you used the RANK or DENSE_RANK functions).
```